### PR TITLE
Reduced gas cost for call to self

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
I'm puzzled. Not sure what the requirements are on the filename, because `eip-X.md` only states
> When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`.

But travis fails with:
```
Files found with invalid names:
eip-draft_reduced_cost_call_self.md
```

**Comment by chriseth** [comment]
I would prefer a more comprehensive solution that also reduces the costs for e.g. repeated calls to the same library.

File: EIPS/eip-1380.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Reducing the call-to-self gas cost would greatly benefit smart contract languages, such as Solidity and Vyper, who would then be able to utilise `CALL` instead
+of `JUMP` opcodes for internal function calls. While languages can already utilise `CALL` for internal function calls, they are discouraged to do so due to the
+gas costs associated with it.
+
+Using `JUMP` comes at a considerable cost in complexity to the implementation of a smart contract language and/or compiler. The context (including stack and memory)
+must be swapped in and out of the calling functions context. A desired feature is having *pure* functions, which do not modify the state of memory, and realising
+them through `JUMP` requires a bigger effort from the compiler as opposed to being able to use `CALL`s.
+
+Using call-to-self provides the guarentee that when making an internal call the function can rely on a clear reset state of memory or context, benefiting both
+contract writers and contract consumers against potentially undetetected edge cases were memory could poison the context of the internal function.
```
Comment (gumb0) [inline EIPS/eip-1380.md]: typo in undetected

**Comment by axic** [comment]
> I would prefer a more comprehensive solution that also reduces the costs for e.g. repeated calls to the same library.

As discussed offline, that would really be similar to how "net sstore gas metering" is working: the VM should maintain the list of loaded contracts by a given contract and only charge once for loading it.

Therefore it doesn't make sense to extend this EIP with that, but would warrant its own EIP.

**Comment by chriseth** [comment]
Note that such an extended version would also reduce the cost of contract calls because the actual `call` after an `extcodesize` check would be cheaper.

**Pull request merged**